Sources/JSONUtilities/Errors.swift

```diff
@@ -16,13 +16,17 @@ import Foundation
 public enum DecodingError: Error, CustomStringConvertible {
   case mandatoryKeyNotFound(key: StringProtocol)
   case mandatoryRawRepresentableHasIncorrectValue(rawRepresentable:Any, rawValue:JSONRawType)
+  case incorrectType(expected: Any.Type, found: Any)
 
   public var description: String {
     switch self {
     case .mandatoryKeyNotFound(let key):
       return "mandatoryKeyNotFound: \(key)"
     case .mandatoryRawRepresentableHasIncorrectValue(let rawRepresentable, let rawValue):
       return "mandatoryRawRepresentableHasIncorrectValue: \(rawRepresentable): \(rawValue)"
+    case .incorrectType(let expected, let recieved):
```

Comment:
probably for consistency the `received` should be `found as declared in the `enum`